src/train.py

```diff
@@ -71,7 +71,10 @@
 
     # Define criterion and optimizer
     criterion = getattr(nn, model_parameters.criterion)
-    criterion = criterion(weight=torch.tensor(model_parameters.weights,dtype=torch.float),
+    # Convert the string to a list of floats
+    weights = [float(x) for x in model_parameters.weights.strip('[]').split(',')]
```

Comment:
`weigths` should be send to device before parsing them to `train_segmentation`

Reply:
Addressed